src/query/atomWithQuery.ts

```diff
@@ -98,16 +98,19 @@ export function atomWithQuery<
         const unsubscribe = observer.subscribe(listener)
         return unsubscribe
       }
-      return { dataAtom, options }
+      return {
+        dataAtom,
+        observer,
+      }
     },
     (get, set, action: AtomWithQueryAction) => {
       switch (action.type) {
         case 'refetch': {
-          const { dataAtom, options } = get(queryDataAtom)
+          const { dataAtom, observer } = get(queryDataAtom)
           set(dataAtom, new Promise<TData>(() => {})) // infinite pending
-          const queryClient = get(getQueryClientAtom)
-          queryClient.getQueryCache().find(options.queryKey)?.reset()
-          const p: Promise<void> = queryClient.refetchQueries(options.queryKey)
+          const p: Promise<void> = observer
+            .refetch({ cancelRefetch: true })
+            .then(() => {})
```

Comment:
If I understand it correctly, either `p: Promise<void> =` or `.then(() => {})` does the hack of returning `Promise<void>`. It doesn't need both.

Reply:
as it is now:
![Screen Shot 2021-07-01 at 7 55 36 AM](https://user-images.githubusercontent.com/11803153/124131558-8c842080-da45-11eb-8b2b-382e40c963bd.png)

no explicit typing:

![Screen Shot 2021-07-01 at 7 55 48 AM](https://user-images.githubusercontent.com/11803153/124131559-8d1cb700-da45-11eb-98bd-ff2bcbab98ab.png)

no .then:

![Screen Shot 2021-07-01 at 8 22 29 AM](https://user-images.githubusercontent.com/11803153/124131560-8d1cb700-da45-11eb-8ba9-ed8d1f0329e6.png)

Reply:
oh, the queryDataAtom is not properly typed... let me fix.